### PR TITLE
Move loading animation to the Animate component

### PR DESCRIPTION
**Pull request by kjellr**
Followup related to #17090.

This PR takes Gutenberg's loading animation and moves it into the Animate component. Previously, this was defined as a mixin in [`_animate.scss`](https://github.com/WordPress/gutenberg/blob/c819d0e0cb8e6a99e3d435150d5572a9edf17431/assets/stylesheets/_animations.scss#L1), with the keyframes declared in three separate places: 

- [packages/edit-widgets/src/style.scss](https://github.com/WordPress/gutenberg/blob/05b9e95a63ff1b278f9d6cfb1d63a84b54d1cbda/packages/edit-widgets/src/style.scss#L56)
- [playground/src/style.scss](https://github.com/WordPress/gutenberg/blob/ef95cb95c2db5b44725c3c2a8778e7098925e170/playground/src/style.scss#L39-L65)
- [packages/edit-post/src/style.scss](https://github.com/WordPress/gutenberg/blob/2d369dca25ff9a2ed9a0c65dc4612ee5ac018be0/packages/edit-post/src/style.scss#L26-L52)

Moving it to the Animate component removes that duplication, and allows developers to include this in a per-component basis. 

This PR: 

- Adds the loading animation to the Animate component, and updates documentation there. 
- Updates the `post-saved-state` and `file` block components to use the Animate component. 
- Removes the (now-unused) animation keyframes and mixin. 

---

## Screenshots

These should look exactly the same as before, but here's what you should see when testing: 

_File Block "Uploading" state:_ 
image

_Saving state:_ 
save

### CODE REVIEW

File: packages/block-library/src/file/edit.js
Comment (kjellr) [inline packages/block-library/src/file/edit.js]: By the way: this seems to work, but it feels wrong to me. This was my first foray into working with this component, so I'd love to learn how to properly get this working here. Any tips are certainly welcome. 😄

**Comment by jasmussen** [comment]
I haven't yet had time to go deep on the animate component, so I think I'll have to defer to someone like Riad who has. But just wanted to say thank you for working on this, and I absolutely think this is the right place for this stuff! 👍 👍

File: packages/block-library/src/file/edit.js
Comment (youknowriad) [inline packages/block-library/src/file/edit.js]: This prop is not needed.

File: packages/block-library/src/file/edit.js
Comment (youknowriad) [inline packages/block-library/src/file/edit.js]:
```suggestion
				<Animate type={ isBlobURL( href ) ? 'loading' : null }>
```

This should be enough here

File: packages/editor/src/components/post-saved-state/index.js
Comment (youknowriad) [inline packages/editor/src/components/post-saved-state/index.js]: Seeing this component used and the fact that it adds a level in the hierarchy makes me wonder if we could replace it with a hook. (Not for this PR though)

```js
const  animationProps = useAnimation( type );
```

Comment (kjellr) [reply packages/editor/src/components/post-saved-state/index.js]: Yeah, I agree — implementing this felt a bit verbose.

**Comment by kjellr** [comment]
Thanks for the help, @youknowriad! Those changes are committed, so I _think_ this should be good to go.

**Review by youknowriad — APPROVED** [review]

**Comment by kjellr** [comment]
@youknowriad any idea what's going on with those failing tests?

**Comment by youknowriad** [comment]
The unit tests failure are legit, the build artifacts one are not (we've been seeing failures the whole day, probably related to an npm upgrade)

**Comment by kjellr** [comment]
Ok, thanks. It looks like that new `<Animate>` wrapper is causing the unit test to fail. I'm not super-familiar with how those tests are constructed, so any help patching that would be much appreciated.

**Comment by gziolo** [comment]
I added 596000d which should unblock this PR :)

**Comment by kjellr** [comment]
🎉 Tests are green. Thanks for the fix, @gziolo!

**Pull request merged**
